src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -386,7 +388,19 @@ public void write() {
             }
           case SHOULD_ROTATE:
             updateRotationTimers(currentRecord);
-            closeTempFile();
+            try {
+              closeTempFile();
+            } catch (Exception e) {
+              log.error("Failed to close temp file.",e);
+              if (this.failedTempFileCloseAttempts < 4) {
+                log.info("Preparing recovery for the "
+                    + "partition attempt : {}",failedTempFileCloseAttempts);
+                startRecovery();
+                failedTempFileCloseAttempts += 1;
```

Comment:
Do this at the start of the exception handling, to indicate that there was a failure.

Reply:
log statement will be moved to the start of the exception handling.